openquake/hazardlib/geo/surface/kite_fault.py

```diff
@@ -0,0 +1,754 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2020 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module :mod:`openquake.hazardlib.geo.surface.kite_fault` defines
+:class:`KiteSurface`.
+"""
+
+import copy
+import numpy as np
+
+from pyproj import Geod
+from openquake.hazardlib.geo.mesh import Mesh
+from openquake.hazardlib.geo import Point, Line
+from openquake.hazardlib.geo.surface.base import BaseSurface
+from openquake.hazardlib.geo.geodetic import npoints_towards
+from openquake.hazardlib.geo.geodetic import distance, azimuth
+
+TOL = 0.5
+TOLERANCE = 0.2
+
+
+class KiteSurface(BaseSurface):
+    """
+    The Kite Fault Surface allows the construction of faults with variable
+    width along the strike, variable dip angle along the dip and strike
+    composed by several disaligned segments. Thrust faults and listric faults
+    can be easily implemented.
+
+    """
+
+    def __init__(self, mesh):
+        self.mesh = mesh
+        assert 1 not in self.mesh.shape, (
+            "Mesh must have at least 2 nodes along both length and width.")
+        # Make sure the mesh respects the right hand rule
+        self._fix_right_hand()
+        self.strike = self.dip = None
+
+    def _fix_right_hand(self):
+        """
+        """
```

Comment:
Private methods should not have a docstring (but may have a comment).

Reply:
Fixed